packages/flutter_tools/test/integration.shard/test_data/migrate_project.dart

```diff
@@ -0,0 +1,237 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+@Timeout(Duration(seconds: 600))
+
+import 'dart:io';
+import 'package:file/file.dart';
+
+import '../../src/common.dart';
+import '../test_utils.dart';
+import 'project.dart';
+
+class MigrateProject extends Project {
+  MigrateProject(this.version, {this.vanilla = true});
+
+  @override
+  Future<void> setUpIn(Directory dir, {
+    bool useDeferredLoading = false,
+    bool useSyntheticPackage = false,
+  }) async {
+    this.dir = dir;
+    _appPath = dir.path;
+    if (androidLocalProperties != null) {
+      writeFile(fileSystem.path.join(dir.path, 'android', 'local.properties'), androidLocalProperties);
+    }
+    final Directory tempDir = createResolvedTempDirectorySync('cipd_dest.');
+    final Directory depotToolsDir = createResolvedTempDirectorySync('depot_tools.');
+
+    await processManager.run(<String>[
+      'git',
+      'clone',
+      'https://chromium.googlesource.com/chromium/tools/depot_tools',
```

Comment:
DBC

This should be pinned, and it should do a shallow clone.

Reply:
For context: https://github.com/flutter/flutter/issues/100254

We already have infrastructure for downloading CIPD artifacts for tests